backend/de.metas.business/src/main/java-legacy/org/compiere/model/MOrder.java

```diff
@@ -461,36 +464,25 @@ public void setBPartner(final I_C_BPartner bp)
 		final IBPartnerDAO bPartnerDAO = Services.get(IBPartnerDAO.class);
 
 		// Set Locations
-		final List<I_C_BPartner_Location> locs = InterfaceWrapperHelper.createList(
-				bPartnerDAO.retrieveBPartnerLocations(bp),
-				I_C_BPartner_Location.class);
-
-		for (final I_C_BPartner_Location loc : locs)
-		{
-			if (loc.isShipTo())
-			{
-				super.setC_BPartner_Location_ID(loc.getC_BPartner_Location_ID());
-			}
-			if (loc.isBillTo())
-			{
-				setBill_Location_ID(loc.getC_BPartner_Location_ID());
-			}
-		}
-		// set to first
-		//No longer setting any location when no shipping location exists for the bpartner
-		if (getBill_Location_ID() == 0 && !locs.isEmpty())
-		{
-			setBill_Location_ID(locs.get(0).getC_BPartner_Location_ID());
-		}
-		if (getC_BPartner_Location_ID() == 0)
+		final BPartnerLocationId shipToLocationId = bPartnerDAO.retrieveBPartnerLocationId(BPartnerLocationQuery.builder()
```

Comment:
further up this method i found `// FIXME: keep in sync / merge with de.metas.adempiere.service.impl.OrderBL.setBPartner(I_C_Order, I_C_BPartner)`

that OrderBL is today in a different package..I moved "fixme" to java-doc-level and also fixed the package-name..

but either way, i think that at least you need to
- add the `bPartnerDAO.retrieveBPartnerLocationId` call there
- add the BPartnerLocationAndCaptureId stuff here.

Reply:
synced.